packages/core/src/RenderingEngine/Viewport.ts

```diff
@@ -990,13 +989,17 @@ class Viewport implements IViewport {
         element: this.element,
         viewportId: this.id,
         renderingEngineId: this.renderingEngineId,
-        rotation: this.rotation,
+        rotation: this.getRotation(),
       };
 
       triggerEvent(this.element, Events.CAMERA_MODIFIED, eventDetail);
     }
   }
 
+  protected getRotation(): number {
+    return 0;
+  }
```

Comment:
I'm lost, what is going on here?

Reply:
Viewport.rotation is not really needed. However, `triggerCameraModifiedEventIfNecessary` needs a rotation and gets it from here. Subclasses that incorporate rotation can override this hook method to provide the rotation.

Reply:
Notice that `StackViewport.getRotation` overrides this and returns `this.rotatiocCache`.

Reply:
I suppose it is also worth mentioning here that in the base `Viewport` class, `this.rotation` was always zero and that is why `getRotation` in the base `Viewport` class returns zero by default.